base/cvd/cuttlefish/host/commands/cvd/fetch/fetch_cvd.cc

```diff
@@ -359,9 +359,10 @@ Result<std::unique_ptr<BuildApi>> GetBuildApi(const BuildApiFlags& flags) {
       StringFromEnv("HOME", ".") + "/.acloud_oauth2.dat";
 
   std::unique_ptr<CredentialSource> credential_source =
-      cvd_creds.ok() ? std::move(*cvd_creds)
-                     : CF_EXPECT(GetCredentialSourceFromFlags(
-                           *retrying_http_client, flags, oauth_filepath));
+      cvd_creds.ok() && cvd_creds->get()
```

Comment:
Given that `Err` and `Ok(nullptr)` are treated in the same way, could `CredentialForScopes` just return `Result<CredentialSource>`, returning an error when it would otherwise return `Ok(nullptr)`?

Reply:
Unfortunately `Result<CredentialSource>` isn't valid because `CredentialSource` is a pure virtual class and therefore cannot be passed by value.

Reply:
Changed `CredentialForScopes` so it doesn't ever return `nullptr`, reducing the space of return values.